python/interpret_community/mimic/models/linear_model.py

```diff
@@ -174,26 +171,240 @@ def predict_proba(self, dataset, **kwargs):
+        :type dataset: numpy or scipy array
+        :return: The predictions of the model.
+        :rtype: list
+        """
+        return super(LogisticRegression, self).predict(dataset, **kwargs)
+
+    def predict_proba(self, dataset, **kwargs):
+        """Call linear predict_proba to predict probabilities using the glassbox model.
+
+        :param dataset: The dataset to predict probabilities on.
+        :type dataset: numpy or scipy array
+        :return: The predictions of the model.
+        :rtype: list
+        """
+        return super(LogisticRegression, self).predict_proba(dataset, **kwargs)
+
+    __init__.__doc__ = (__init__.__doc__ +
+                        '\nUses the parameters for LogisticRegression:\n' +
+                        SKLogisticRegression.__doc__.replace('-', ''))
+
+    fit.__doc__ = (fit.__doc__ +
+                   '\nUses the parameters for LogisticRegression:\n' +
+                   SKLogisticRegression.fit.__doc__.replace('-', ''))
+
+    predict.__doc__ = (predict.__doc__ +
+                       '\nUses the parameters for LogisticRegression:\n' +
+                       SKLogisticRegression.predict.__doc__.replace('-', ''))
+
+    predict_proba.__doc__ = (predict_proba.__doc__ +
+                             '\nUses the parameters for LogisticRegression:\n' +
+                             SKLogisticRegression.predict_proba.__doc__.replace('-', ''))
+
+
+class LinearExplainableModel(LinearClassifierMixin, LinearBase, BaseExplainableModel):
+    available_explanations = [Extension.GLOBAL, Extension.LOCAL]
+    explainer_type = Extension.GLASSBOX
+
+    """Linear explainable model.
+
+    :param multiclass: Set to true to generate a multiclass model.
+    :type multiclass: bool
+    :param random_state: Int to seed the model.
+    :type random_state: int
+    """
+
+    def __init__(self, multiclass=False, random_state=DEFAULT_RANDOM_STATE, classification=True, **kwargs):
+        """Initialize the LinearExplainableModel.
+
+        :param multiclass: Set to true to generate a multiclass model.
+        :type multiclass: bool
+        :param random_state: Int to seed the model.
+        :type random_state: int
+        """
+        self.multiclass = multiclass
+        self.random_state = random_state
+        if self.multiclass:
+            initializer = SKLogisticRegression
+            kwargs['random_state'] = random_state
+        else:
+            initializer = SKLinearRegression
+        super(LinearExplainableModel, self).__init__(initializer, **kwargs)
+        self._logger.debug('Initializing LinearExplainableModel')
+        self._method = 'mimic.linear'
+        self._linear_explainer = None
+        self._classification = classification
+
+    def fit(self, dataset, labels, **kwargs):
+        """Call linear fit to fit the explainable model.
+
+        :param dataset: The dataset to train the model on.
+        :type dataset: numpy or scipy array
```

Comment:
do we take pandas dataframes? If not, that's bad

Reply:
not in the explainable model, but in the classifier/regressor we should be able to handle it.  let me add a test case.